client/components/phone-input/country-flag.jsx

```diff
@@ -5,68 +5,60 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { Component } from 'react';
 import Gridicon from 'gridicons';
-
 /** Internal Dependencies */
 import Spinner from 'components/spinner';
 
-export default class extends React.Component {
+export default class extends Component {
 	static displayName = 'PhoneInputCountryFlag';
 
 	static propTypes = {
 		countryCode: PropTypes.string.isRequired,
 	};
 
 	state = {
-		ready: false,
-		error: false,
+		flagSvg: null,
 	};
 
-	componentDidUpdate( oldProps ) {
-		if ( this.props.countryCode && this.props.countryCode !== oldProps.countryCode ) {
-			this.setState( { ready: false, error: false } );
-		}
+	componentDidMount() {
+		this.updateFlag( null, this.props.countryCode );
 	}
 
-	renderSpinner = () => {
-		if ( ( ! this.props.countryCode || ! this.state.ready ) && ! this.state.error ) {
-			return <Spinner size={ 16 } className="phone-input__flag-spinner" />;
-		}
-	};
-
-	handleImageLoad = () => {
-		this.setState( { ready: true, error: false } );
-	};
-
-	handleImageError = () => {
-		this.setState( { ready: false, error: true } );
-	};
+	componentDidUpdate( prevProps ) {
+		this.updateFlag( prevProps.countryCode, this.props.countryCode );
+	}
 
-	renderFlag = () => {
-		const style = this.state.ready ? {} : { visibility: 'hidden' };
+	updateFlag = ( prevCode, nextCode ) => {
+		if ( prevCode !== nextCode ) {
```

Comment:
Can we move this check into `componentDidUpdate()`? That seems to be the only place it's really needed (same goes for the `prevCode` arg).

Reply:
if we move this into `componentDidUpdate()` then it won't run on component mount. we need it to load the initial flag if this component mounts with a specified flag